### ISSUE TITLE
[bug] LSP in 0.25.0 does not work if config file is outside the workspace

### ISSUE THREAD

**Issue by kam193** [ast-grep/ast-grep#1326]
# ⏯ Playground Link

[Playground link with relevant code](https://ast-grep.github.io/playground.html)

# 💻 Code

Code:

```
exec("something")
```

Pattern:

```yml
id: example
message: Using exec in code
severity: error
language: Python
rule:
  pattern: exec($CODE)
```

Configuration: I use the LSP pragmatically in a project, but to verify the problem, I configured a similar environment in the VSCode using the official extension.

Directory stucture is like:
```
| rules/
| | sgconfig.yaml
| | python/
| | | rules.yaml
| src/
| | code.py
```

LSP was configured with the absolute path to the `sgconfig.yaml`.

# 🙁 Actual behavior

After updating to 0.25.0 (from PyPI), the LSP server does not work if the config file is not in the workspace root. Rules are loaded (I've checked, pointing to wrong paths triggers an error), but no diagnostic notes are provided. My suspicion is that the fix for #1272 tries to get the path of the analysed file relative to the config, without ensuring the file _is_ after the config in the directory tree.

Screens from VSCode using 0.25.0: 
![obraz](https://github.com/user-attachments/assets/61618b07-fc16-479a-94d8-457e4b759c93)
![obraz](https://github.com/user-attachments/assets/cc5b9549-6f29-4239-bb91-8d8b5fa48250)

# 🙂 Expected behavior

If using the version 0.24.1 (from PyPI) **or** moving `sgconfig.yaml` to the workspace root (adjusting paths to rules), everything works as expected:

![obraz](https://github.com/user-attachments/assets/c5a2e77c-9a1c-4d5e-825c-a6e9010f2d36)

**Comment by kam193** [comment]
Thank you for the fix!

**Comment by HerringtonDarkholme** [comment]
released in 0.25.1. it should work now.
